Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -316,33 +498,25 @@ extension HTTPHeaders: Sequence {
     /// This iterator will return each value for a given header name separately. That
     /// means that `name` is not guaranteed to be unique in a given block of headers.
     public struct Iterator: IteratorProtocol {
-        private var storageIterator: HTTPHeadersStorage.Iterator
-        private var valuesIterator: Array<(String, String)>.Iterator?
+        private let storage: HTTPHeadersByteStorage
+        private var indiciesIt: Array<(HTTPHeaderIndex)>.Iterator
 
-        fileprivate init(wrapping: HTTPHeadersStorage.Iterator) {
-            self.storageIterator = wrapping
+        fileprivate init(storage: HTTPHeadersByteStorage) {
+            self.storage = storage
+            indiciesIt = storage.indicies.makeIterator()
         }
 
         public mutating func next() -> Element? {
-            // If we're already iterating an entry in the dict, grab the next one
-            if let nextValues = valuesIterator?.next() {
-                return nextValues
-            } else {
-                // If there's nothing left in this array, clear the iterator
-                valuesIterator = nil
-            }
-
-            if let entry = storageIterator.next() {
-                valuesIterator = entry.value.makeIterator()
-                return next()
-            } else {
-                return nil
+            if let nameIdx = indiciesIt.next() {
+                let valueIdx = indiciesIt.next()!
+                return (storage.string(idx: nameIdx), storage.string(idx: valueIdx))
             }
+            return nil
         }
     }  
 
     public func makeIterator() -> Iterator {
-        return Iterator(wrapping: storage.makeIterator())
```

Comment:
Why change this label?

Reply:
No idea... I like storage more. That said I can revert.